src/plugins/badge.coffee

```diff
@@ -0,0 +1,620 @@
+
+    if @parent_doc.addEventListener # Mozilla, Opera, Webkit
+      @parent_doc.addEventListener 'DOMContentLoaded', cb, false
+    else if @parent_doc.attachEvent # IE
+      @parent_doc.attachEvent('onreadystatechange', (=> cb() if @parent_doc.readyState == 'complete'))
+    else
+      setTimeout((=> try cb() catch), DOMREADY_TIMEOUT)
+
+  ###
+  Displays the plugin
+  ###
+  render: ->
+    switch context().configuration.display
+      when 'floating' then @_render_floating()
+      when 'embedded' then @_render_embedded()
+
+  _log_error: (message) ->
+    console.log("Skoutz Partner Badge: ERROR: " + message)
+
+  _render_floating: ->
+    @$el = document.createElement('div')
+    @$el.id = 'sa-badge-floating-plugin'
+    @$el.innerHTML = FLOATING_TEMPLATE(stars: @rating_to_stars(context().data.rating))
+    @parent_doc.body.appendChild(@$el)
+
+  _render_embedded: ->
+    @$el = @$embeddedBadgePlugin()
+    rating = context().data.rating
+    @$el.innerHTML = EMBEDDED_TEMPLATE(rating: rating, stars: @rating_to_stars(rating))
+
+  $embeddedBadgePlugin: -> @parent_doc.getElementById('sa-badge-embedded-plugin')
+  $embeddedBadgeMoreButton: -> document.getElementById('sa-badge-embedded-more-button')
+  $closeModalButton: -> document.getElementById('sa-badge-modal-close-button')
+  $modalIframe: -> document.getElementById("sa-badge-modal-iframe")
+  $modalSpinner: -> document.getElementById("sa-badge-modal-spinner")
+
+  ###
+  Adds the required css for the plugin
+  ###
+  addStyle: ->
+    style = document.createElement('style')
+    style.id = 'sa-badge-style'
+    style.type = 'text/css'
+
+    # IE compatibility
+    if style.styleSheet
+      style.styleSheet.cssText = STYLE(context().configuration)
+    else
+      style.appendChild(document.createTextNode(STYLE(context().configuration)))
+
+    @_head().appendChild(style)
+
+  _bindHandlers: ->
+    if context().configuration.embedded and @$embeddedBadgePlugin()
+      @_attachEvent(@$embeddedBadgeMoreButton(), 'click', @_onClickEmbeddedBadgeMoreButton)
+    else
+      @_attachEvent(@$el, 'click', @_onClickBadge)
+
+  _attachEvent: (element, event, handler) ->
+    return element.addEventListener event, handler, false  if element.addEventListener
+    element.attachEvent "on#{event}", handler              if element.attachEvent
+
+  _hideModal: ->
+    return unless @$modal?
+
+    @$modal.style.display = "none"
+
+  _onClickModalCloseButton: =>
+    @_hideModal()
+
+  _createModal: ->
+    @$modal = document.createElement('div')
+    @$modal.id = 'sa-badge-modal'
+    @$modal.innerHTML = MODAL_INNER_TEMPLATE(endpoint: @_endpoint())
+    @parent_doc.body.appendChild(@$modal)
+    @_attachEvent(@$closeModalButton(), 'click', @_onClickModalCloseButton)
+    @_attachEvent(@$modal, 'click', @_onClickModal)
+    @_attachEvent(document, 'keyup', @_onKeyUp)
+
+  _onClickBadge: =>
+    return @_createModal() unless @$modal?
+
+    @$modal.style.display = 'block'
```

Comment:
I think we should extract this logic to a method as we already do with `_hideModal`, we can add a `_showModal`